### PR TITLE
Adds a stealthdrop example

### PR DESCRIPTION
**Pull request by signorecello**

### CODE REVIEW

**Comment by signorecello** [comment]
Ready for review @critesjosh

**Comment by critesjosh** [comment]
Should you rebase this branch? It looks like it is missing a few of the examples that are in main.

File: ec/src/main.nr
Comment (critesjosh) [inline ec/src/main.nr:1]: Should we put circuits and related files (`Prover.toml`, `Nargo.toml`, etc) into a circuits directory like we do in the Foundry project examples?

I think having consistency across projects will help with creating familiarity and legibility of Noir projects.

**Comment by signorecello** [comment]
done @critesjosh :) also removed barretenberg and other dependencies causing trouble that aren't needed anymore

**Comment by critesjosh** [comment]
You mentioned bb.js doesn't play nice with hardhat. Can we move this project away from being a hardhat project? 

It looks like the only hardhat thing that is used is `npx hardhat test` which could just be replaces with `yarn test`, is that right?

**Comment by critesjosh** [comment]
Also consider updating the README at the project root to include this.

**Comment by signorecello** [comment]
We also use hardhat to compile the circuits... I'll see if there's something we can replace it with, maybe compiling directly with solc or something

**Pull request merged**